catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/QueryMonitorPlugin.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.querymonitor;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+import ddf.catalog.plugin.PostFederatedQueryPlugin;
+import ddf.catalog.plugin.PreFederatedQueryPlugin;
+
+/**
+ * A {@link QueryMonitorPlugin} monitors ActiveSearches by maintaining a {@link HashMap} known as the SearchTable with a key of
+ * {@link UUID} and a value of {@link ActiveSearch}. {@link QueryMonitorPlugin} extends both {@link PreFederatedQueryPlugin}
+ * and {@link PostFederatedQueryPlugin} in order to process each search for every source. The {@link PreFederatedQueryPlugin}
+ * should be implemented to add the {@link ActiveSearch} to the SearchTable and the {@link PostFederatedQueryPlugin} should
+ * be implemented to remove the {@link ActiveSearch} from the SearchTable.
+ */
+
+public interface QueryMonitorPlugin<T> extends PreFederatedQueryPlugin, PostFederatedQueryPlugin {
+
+    /**
+     * Returns a copy of the ActiveSearchTable HashMap
+     *
+     * @return HashMap<UUID,ActiveSearch> copy of the ActiveSearchTable HashMap
+     */
+
+    HashMap<UUID, ActiveSearch> getSearchTable();
```

Comment:
HashMap should probably just be Map so you aren't forcing a specific implementation on classes that implement this interface.